crates/nu-command/src/filesystem/save.rs

```diff
@@ -104,6 +104,17 @@ impl Command for Save {
                     res
                 }
             }
+            PipelineData::ListStream(ls, _) => {
+                let (mut file, _) = get_files(&path, &stderr_path, append, force)?;
+                for val in ls {
+                    file.write_all(&value_to_bytes(val)?)
+                        .map_err(|err| ShellError::IOError(err.to_string()))?;
+                    file.write_all("\n".as_bytes())
+                        .map_err(|err| ShellError::IOError(err.to_string()))?;
```

Comment:
Not sure if this is the correct behavior if we want to generate a file in a streaming way for some file types?
This change would always use the `Value::as_string()` if I understand it. But formats like `nuon`/`csv`/`json` are reasonable to stream from a list of values, with some effort.

Reply:
Yeah, this is a regression. Previously `ls | save foo.json` converted `ls` results to JSON, now it fails with `Can't convert to string`.

Maybe that should be fixed. But to be honest I don't like the existing functionality where `save` magically converts formats based on the file extension, and I would not be sad if someone decided to get rid of it.

Reply:
> But formats like nuon/csv/json are reasonable to stream from a list of values, with some effort.

Ah, that's a good catch.

> Maybe that should be fixed. But to be honest I don't like the existing functionality where save magically converts formats based on the file extension, and I would not be sad if someone decided to get rid of it.

I think we should keep this behavior, we have `--raw` flag to control it

Reply:
Currently it will only stream data if `--raw` flag is provided or target file doesn't have an extension

Reply:
Sorry not sure if I consider this resolved.

Reply:
@sholderbach My bad, sorry for merging.

What exactly are you looking for? `ls | save foo.json` now converts to JSON as before.

Reply:
Yeah, I think for something like `ls | save foo.json`, the data is converted to json, streaming only happened when we provide `--raw` flag